core/src/main/java/org/springframework/security/access/AccessDeniedException.java

```diff
@@ -24,6 +27,8 @@
  */
 public class AccessDeniedException extends RuntimeException {
 
+	protected final MessageSourceAccessor messages = SpringSecurityMessageSource.getAccessor();
```

Comment:
@sjohnr Thanks for the feedback, working on these changes. Do you have any thoughts on the below class naming and usage.
`class MessageSourceAccessProvider {

    private MessageSourceAccessProvider(){};

    private static class MessageSourceAccessorHolder {
        static final MessageSourceAccessor INSTANCE = SpringSecurityMessageSource.getAccessor();
    }

    public static MessageSourceAccessor getInstance() {
        return MessageSourceAccessorHolder.INSTANCE;
    }
}`

Reply:
The class looks good. Since we're just focusing on access denied right now, how about `AccessDeniedMessageSourceHolder`? (`Provider` is nice too, so it could work.)

Might want to make the `getInstance()` package-private as well. Not sure it matters, but seems best for consistency.

Reply:
@sjohnr indeed AccessDeniedMessageSourceHolder make more sense.